app/src/main/java/app/revanced/manager/ui/viewmodel/PatcherViewModel.kt

```diff
@@ -63,62 +60,62 @@ class InstallerViewModel(
     private val installedAppRepository: InstalledAppRepository by inject()
     private val rootInstaller: RootInstaller by inject()
 
+    private var installedApp: InstalledApp? = null
     val packageName: String = input.selectedApp.packageName
+    var installedPackageName by mutableStateOf<String?>(null)
+        private set
+    var isInstalling by mutableStateOf(false)
+        private set
+
     private val tempDir = fs.tempDir.resolve("installer").also {
         it.deleteRecursively()
         it.mkdirs()
     }
-
-    private val outputFile = tempDir.resolve("output.apk")
     private var inputFile: File? = null
-
-    private var installedApp: InstalledApp? = null
-    var isInstalling by mutableStateOf(false)
-        private set
-    var installedPackageName by mutableStateOf<String?>(null)
-        private set
-    val appButtonText by derivedStateOf { if (installedPackageName == null) R.string.install_app else R.string.open_app }
+    private val outputFile = tempDir.resolve("output.apk")
 
     private val workManager = WorkManager.getInstance(app)
-
-    private val _progress: MutableStateFlow<ImmutableList<Step>>
-    private val patcherWorkerId: UUID
     private val logger = ManagerLogger()
 
-    init {
-        // TODO: navigate away when system-initiated process death is detected because it is not possible to recover from it.
-
-        viewModelScope.launch {
-            installedApp = installedAppRepository.get(packageName)
-        }
-
-        val (selectedApp, patches, options) = input
+    val patchesProgress = MutableStateFlow(Pair(0, input.selectedPatches.flatMap { it.value }.size))
```

Comment:
This is likely faster because it avoids creating an intermediary list.

```suggestion
    val patchesProgress = MutableStateFlow(Pair(0, input.selectedPatches.values.sum { it.size }))
```

Reply:
`.sum` only seems to work on a list of numbers, i don't think this makes any performance difference since its only run once

Reply:
Seems like its called `sumOf` for Iterables instead of `sum`. I feel like the `sumOf` would communicate what we are trying to do better than the `flatMap`.